django_filters/filters.py

```diff
@@ -478,7 +479,15 @@ def __init__(self, choices=None, filters=None, *args, **kwargs):
         if filters is not None:
             self.filters = filters
 
-        unique = set([x[0] for x in self.choices]) ^ set(self.filters)
+        all_choices = list(
+            chain.from_iterable(
+                [subchoice[0] for subchoice in choice[1]]
+                if isinstance(choice[1], list | tuple)  # This is an optgroup
+                else [choice[0]]
+                for choice in self.choices
+            )
+        )
+        unique = set(all_choices) ^ set(self.filters)
```

Comment:
Is this Python 3.10+? 

Happy to drop Python 3.7 now, but we should be Python 3.8 compatible still.

Reply:
I didn't change this line so I guess it's compatible with all the versions.

The CI problem concerns the `list | tuple` that is not available before Python 3.10.

I'll fix it!

Reply:
Ah, good. Yes. 👍